src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -94,13 +94,9 @@ private void OnItemsChanged(IObservableVector<object> sender, IVectorChangedEven
 
 		protected override void OnItemsSourceChanged(DependencyPropertyChangedEventArgs e)
 		{
-			// Calling this method before base.OnItemsSourceChanged() ensures that, in the case of an ObservableCollection, the list
-			// subscribes to CollectionChanged before AutoSuggestBox does. This is important for Android because the list needs to
-			// notify RecyclerView of collection changes before UpdateSuggestionList() measures it, otherwise we get errors like
-			// "Inconsistency detected. Invalid view holder adapter position"
-			UpdateSuggestionList();
-
 			base.OnItemsSourceChanged(e);
```

Comment:
Changing this will likely break android, which was fixed at that time: https://github.com/unoplatform/uno/pull/1069. You can validate the behavior of this particular change with the android binary and the test provided here:

https://github.com/unoplatform/uno/blob/eee3d255b2ccac1adc311c61d4d59fc2ee1b2df7/src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/BasicAutoSuggestBox.xaml#L1

Reply:
Currently, I don't have the environment that I can test with Android. How about adding one more after base.OnItemsSourceChanged(e)?

```
UpdateSuggestionList();
base.OnItemsSourceChanged(e);
UpdateSuggestionsList();
```
Anyway, UpdateSuggestionsList() have to be after `base.OnItemSourceChanged(e);`

Reply:
When I build SamplesApp.Droid, I have some error related with Xamarainmac20.
How can I fix it? Can I remove the Xamarainmac20 in TargetPlatform of all the .csproj to fix?

Reply:
@TopProgrammer77 Which solution filter you are dealing with? What UnoTargetFrameworkOverride value are you using? and what is the error?

Reply:
@Youssef1313 ,Where to set the UnoTargetFrameworkOverride?

Reply:
> Changing this will likely break android, which was fixed at that time: #1069. You can validate the behavior of this particular change with the android binary and the test provided here:
> 
> https://github.com/unoplatform/uno/blob/eee3d255b2ccac1adc311c61d4d59fc2ee1b2df7/src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/BasicAutoSuggestBox.xaml#L1

@jeromelaban , I tested in Android and the error was not detected. The error is resolved by @Xiaoy312 in https://github.com/unoplatform/uno/blob/3be20e489d67ed2c024163173f492ccb82566fbc/src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs#L688

Reply:
The android issue was related with https://github.com/unoplatform/uno/pull/12158. By merging this [PR](https://github.com/unoplatform/uno/pull/12158), it was resolved. So I think it is right to call `UpdateSuggestionList()` after `base.OnItemsSourceChanged(e)`.

Reply:
It's generally best to avoid resolving comments until the maintainers have actually provided the resolution.

I've been able to test the changes on a device, and it is working properly. Thanks for the research!